### PR TITLE
Improve documentation for UPF installation

### PR DESCRIPTION
**Pull request by gab-arrobo**

### CODE REVIEW

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by thakurajayL** [comment]
ok to test

File: docs/INSTALL.md [hunk 1]
```diff
@@ -3,35 +3,226 @@ SPDX-License-Identifier: Apache-2.0
...
+    diff --git a/conf/upf.json b/conf/upf.json
+    index c6531cc..335e33b 100644
+    --- a/conf/upf.json
+    +++ b/conf/upf.json
+    @@ -51,7 +51,7 @@
+         "gtppsc": false,
+
+         "": "Enable Intel Dynamic Device Personalization (DDP)",
+    -    "ddp": false,
+    +    "ddp": true,
```
Comment (thakurajayL) [inline docs/INSTALL.md:120]: Why this is mandatory for dpdk ?

File: docs/INSTALL.md [hunk 2]
```diff
@@ -3,35 +3,226 @@ SPDX-License-Identifier: Apache-2.0
...
+
+         "": "Telemetrics-See this link for details: https://github.com/NetSys/bess/blob/master/bessctl/module_tests/timestamp.py",
```
Comment (thakurajayL) [inline docs/INSTALL.md:122]: Does this change now given we have bess repository forked ?

File: docs/dpdk-configuration.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Copyright 2022 Intel Corporation
+-->
+
+# DPDK Configuration
+
+If you are planning to use the UPF with DPDK, the following steps are required to properly configure the devices:
+
+- Clone `dpdk`
+```bash
+$ git clone https://github.com/DPDK/dpdk.git
```
Comment (thakurajayL) [inline docs/dpdk-configuration.md:12]: Are you sure about latest dpdk version ? I though Amar was saying that bess is tied up with specific version of BESS

Comment (gab-arrobo) [reply docs/dpdk-configuration.md:12]: Yes, BESS works with dpdk-20.11.3. However, this step is just to bind the devices. It does not mean that BESS will use this (latest) version of DPDK

File: docs/INSTALL.md (hunk 2 above)
Comment (gab-arrobo) [reply docs/INSTALL.md:122]: Yes, I am going to add another commit changing all the references to the new repo/fork.

Comment (gab-arrobo) [reply docs/INSTALL.md]: I just added another commit to this PR to address your comment. However, there are still two places in the code with links to the original BESS. One is in `Dockerfile` that is address in #637 and in `pfcpiface/bess_pb/module_msg.pb.go` that needs to be changed/updated in the BESS fork and regenerate the GO code with the updated links

File: docs/INSTALL.md (hunk 1 above)
Comment (amarsri28) [inline docs/INSTALL.md:120]: ddp mode in intel specific mode and not a mandatory requirement for sriov/dpdk, setting it true could result in failing on other vendors card

File: docs/dpdk-configuration.md (hunk 3 above)
Comment (amarsri28) [reply docs/dpdk-configuration.md:12]: hey Gabriel, its better to mention actual version of dpdk we prefer for binding as there is no guarantee of always having backward compatibility of dpdk latest vesrion, even for binding of devices and other operations.

File: docs/INSTALL.md [hunk 4]
```diff
@@ -3,35 +3,226 @@ SPDX-License-Identifier: Apache-2.0
...
+    +ifaces=("access" "core")
+
+     # Static IP addresses of gateway interface(s) in cidr format
+     #
+    @@ -32,7 +32,7 @@ ipaddrs=(198.18.0.1/30 198.19.0.1/30)
+     # MAC addresses of gateway interface(s)
+     #
+     # In the order of (s1u sgi)
+    -macaddrs=(9e:b2:d3:34:ab:27 c2:9c:55:d4:8a:f6)
+    +macaddrs=(b4:96:91:b1:ff:f0 b4:96:91:b1:ff:f1)
```
Comment (amarsri28) [inline docs/INSTALL.md:163]: got bit confused after proposed changes in macaddrs/others , but when i saw dpdk config, i could link it up , but it wont be easy for newcomers/first time setters to understand what are the specific parameters required from dpdk, appreciate if we can specifically mention - i.e macaddrs, dev etc need to be derived by runnin dpdk tools/others etc.

File: docs/dpdk-configuration.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
+<!--
+SPDX-License-Identifier: Apache-2.0
+Copyright 2022 Intel Corporation
+-->
+
```
Comment (amarsri28) [inline docs/dpdk-configuration.md:5]: lot of users faced issue as they missed on prerequisite requirements for enabling virtualization, without that this dpdk script/tools wont produce output given below. pls mention those as well

File: docs/INSTALL.md [hunk 6]
```diff
@@ -88,61 +269,7 @@ without the need for external interfaces.
 
 ![ubench-sim](images/ubench-sim.svg)
 
-To start UPF in simulation mode:
```
Comment (amarsri28) [inline docs/INSTALL.md:91]: reference removed for enabling sim mode , is it as by default its enable. what if user want to switch from dpdk to sim mode. some developers faced this issue and ask for help earlier and this reference helped them

File: docs/INSTALL.md [hunk 7]
```diff
@@ -88,61 +269,7 @@ without the need for external interfaces.
...
-    +++ b/conf/upf.json
-    @@ -2,7 +2,7 @@
-        "": "Vdev or sim support. Enable `\"mode\": \"af_xdp\"` to enable AF_XDP mode, or `\"mode\": \"af_packet\"` to enable AF_PACKET mode, or `\"mode\": \"sim\"` to generate synthetic traffic from BESS's Source module",
-        "": "mode: af_xdp",
-        "": "mode: af_packet",
-    -    "": "mode: sim",
-    +    "mode": "sim",
-
-        "": "max UE sessions",
-        "max_sessions": 50000,
```
Comment (amarsri28) [inline docs/INSTALL.md:108]: sim mode is good for exploration, one can play with parameters like max_session, tables_sizes and immediately verify upf functioning in standalone mode without requiring hardware/aiab etc. we shall mention these capabilities to users who want explore upf from their point of view

**Review by amarsri28 — COMMENTED** [review]
review

File: docs/dpdk-configuration.md (hunk 3 above)
Comment (gab-arrobo) [reply docs/dpdk-configuration.md:12]: > its better to mention actual version of dpdk we prefer for binding as there is no guarantee of always having backward compatibility of dpdk latest vesrion, even for binding of devices and other operations.

I do not understand your comment. The device binding has nothing to do with BESS and that should be transparent to the DPDK version used for the binding

File: docs/INSTALL.md (hunk 1 above)
Comment (gab-arrobo) [reply docs/INSTALL.md]: Good point. Removed

Comment (gab-arrobo) [reply docs/INSTALL.md]: Good point. Removed

File: docs/INSTALL.md [hunk 8]
```diff
@@ -3,35 +3,226 @@ SPDX-License-Identifier: Apache-2.0
...
+            diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+    index f5946c1..637d62b 100755
+    --- a/scripts/docker_setup.sh
+    +++ b/scripts/docker_setup.sh
+    @@ -22,7 +22,7 @@ mode="dpdk"
+     # Gateway interface(s)
+     #
+     # In the order of ("s1u" "sgi")
+    -ifaces=("ens803f2" "ens803f3")
+    +ifaces=("access" "core")
```
[the remark this answers is not in the document]
Comment (gab-arrobo) [reply docs/INSTALL.md:145]: @amarsri28, what do you mean with "by running scripts/tools"? what scripts/tools are you referring to? How I see these two names (`access` and `core`) is that they are easy to remember/differentiate which interface points to (is connected to) the `access` side (e.g., N3) and the `core` side (e.g., N6). The important part of this is that these two names here (scripts/docker_setup.sh) have to match with the names in the `conf/upf.json` file

File: docs/INSTALL.md (hunk 4 above)
Comment (gab-arrobo) [reply docs/INSTALL.md:154]: This is why I added the `note` right after step 2. I am going to add a bit more text there to be more explicit
`> NOTE: Please remember to follow [these](dpdk-configuration.md) directions before executing the next step`

File: docs/dpdk-configuration.md (hunk 5 above)
Comment (gab-arrobo) [reply docs/dpdk-configuration.md:5]: Good point. I am going to add that

File: docs/INSTALL.md [hunk 9]
```diff
@@ -3,35 +3,226 @@ SPDX-License-Identifier: Apache-2.0
...
+    +nhmacaddrs=(b4:96:91:b4:47:b8 b4:96:91:b4:47:b9)
+
+     # IPv4 route table entries in cidr format per port
+     #
+    @@ -111,7 +111,7 @@ sudo rm -rf /var/run/netns/pause
+     make docker-build
+
+     if [ "$mode" == 'dpdk' ]; then
+    -       DEVICES=${DEVICES:-'--device=/dev/vfio/48 --device=/dev/vfio/49 --device=/dev/vfio/vfio'}
+    +       DEVICES=${DEVICES:-'--device=/dev/vfio/184 --device=/dev/vfio/185 --device=/dev/vfio/vfio'}
```
[the remark this answers is not in the document]
Comment (gab-arrobo) [reply docs/INSTALL.md:172]: Same as in my other comment... This is why I added the note right after step 2. I am going to add a bit more text there to be more explicit
> NOTE: Please remember to follow [these](dpdk-configuration.md) directions before executing the next step

File: docs/INSTALL.md (hunk 6 above)
Comment (gab-arrobo) [reply docs/INSTALL.md:91]: Added some text to clarify that.

File: docs/INSTALL.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (gab-arrobo) [reply docs/INSTALL.md]: @amarsri28 I made one more commit where I removed the need to change/update the interface names. So, this comment is not applicable anymore.

File: docs/INSTALL.md (hunk 7 above)
Comment (gab-arrobo) [reply docs/INSTALL.md:108]: Well, this document is supposed to be about `installation`. I am going to try to add some lines about your comment in the installation documentation

**Review by thakurajayL — APPROVED** [review]

**Pull request merged**